public/chirpy.js

```diff
@@ -25,32 +26,30 @@ function start () {
-                        rapi.start(agent.kbId, function(data) {
-                            if (goal.subjectInstance === 'user provided' || goal.objectInstance === 'user provided') {
-                                if (goal.subject) {
-                                    addRBChatLine(goal.subject + '?');
-                                } else {
-                                    addRBChatLine(goal.object + '?');
-                                }
-                                d3.select('#sendButton').on('click', function() {
-                                    addUserChatLine(d3.select('#userInput').property('value'));
-                                    var ourQuery = {
-                                        subject: goal.subjectInstance ? d3.select('#userInput').property('value') : null,
-                                        relationship: goal.rel,
-                                        object: goal.objectInstance ? d3.select('#userInput').property('value')  : null
-                                    };
-                                    rapi.query(ourQuery, handleResponse);
-                                    clearUserInput();
-                                });
+                        if (goal.subjectInstance === 'user provided' || goal.objectInstance === 'user provided') {
+                            if (goal.subject) {
+                                addRBChatLine(goal.subject + '?');
                             } else {
+                                addRBChatLine(goal.object + '?');
+                            }
+                            d3.select('#sendButton').on('click', function() {
+                                addUserChatLine(d3.select('#userInput').property('value'));
                                 var ourQuery = {
-                                    subject: goal.subjectInstance ? goal.subjectInstance : null,
+                                    subject: goal.subjectInstance ? d3.select('#userInput').property('value') : null,
                                     relationship: goal.rel,
-                                    object: goal.objectInstance ? goal.objectInstance : null
+                                    object: goal.objectInstance ? d3.select('#userInput').property('value')  : null
                                 };
                                 rapi.query(ourQuery, handleResponse);
-                            }
-                        });
+                                clearUserInput();
+                            });
+                        } else {
+                            var ourQuery = {
+                                subject: goal.subjectInstance ? goal.subjectInstance : null,
+                                relationship: goal.rel,
+                                object: goal.objectInstance ? goal.objectInstance : null
+                            };
+                            rapi.query(ourQuery, handleResponse);
```

Comment:
Could this be done just once below the if statement? ourQuery declared just before if statement with goal.rel being common between the two?